### PR TITLE
Allow editing team for OnCall resources

### LINKED ISSUE #854: Update team id at oncall resources are not effective
### Terraform Version
* Terraform: 1.4.1
* Terraform Grafana Provider: 1.36.1
* Grafana: cloud

### Affected Resource(s)
Please list the resources as a list, for example:
* grafana_oncall_escalation_chain
* grafana_oncall_on_call_shift
* grafana_oncall_schedule

### Terraform Configuration Files
```hcl
data "grafana_oncall_team" "team" {
  name = "Switch team"
}

resource "grafana_oncall_on_call_shift" "platform_oncall" {
  team_id  = data.grafana_oncall_team.team.id
 ....
}
```

### Expected Behavior
Force replace resources or update the team.

### Actual Behavior

When updating the team id it never updates.
The plan keeps changing on every run and the resource is never updated with the new team.

```
  # grafana_oncall_schedule.platform_oncall will be updated in-place
  ~ resource "grafana_oncall_schedule" "platform_oncall" {
        id                 = "SLIWJS4X2UWCE"
        name               = "Platform"
      + team_id            = "TNW1ZIT5HWI1Y"
        # (4 unchanged attributes hidden)
    }
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### PR DESCRIPTION
**Pull request by iskhakov**

### CODE REVIEW

File: internal/resources/oncall/resource_integration.go [hunk 1]
```diff
@@ -275,11 +275,13 @@ func ResourceIntegrationUpdate(ctx context.Context, d *schema.ResourceData, m in
 	client := m.(*common.Client).OnCallClient
 
 	nameData := d.Get("name").(string)
+	TeamIDData := d.Get("team_id").(string)
 	templateData := d.Get("templates").([]interface{})
 	defaultRouteData := d.Get("default_route").([]interface{})
 
 	updateOptions := &onCallAPI.UpdateIntegrationOptions{
 		Name:         nameData,
+		TeamID:       TeamIdData,
```
Comment (inkel) [inline internal/resources/oncall/resource_integration.go]: Typo:
```suggestion
		TeamID:       TeamIDData,	
```

Also, let's not use a capital `T` for it.

**Review by inkel — CHANGES_REQUESTED** [review]
I've found a typo which should be corrected. While at it, while not naming the variable `teamID` instead? I know there are others that have a `Data` suffix and I don't like those either, but changing them is out of the scope of this PR 😬

File: internal/resources/oncall/resource_integration.go (hunk 1 above)
Comment (iskhakov) [reply internal/resources/oncall/resource_integration.go]: Thank you for the review! Good catch. All fixed now

**Review by inkel — APPROVED** [review]

**Review by inkel — APPROVED** [review]

**Comment by julienduchesne** [comment]
Hey @iskhakov. Any progress on this? I see it's missing https://github.com/grafana/amixr-api-go-client/pull/11

**Pull request merged**